lib/ae_mdw/dex.ex

```diff
@@ -82,6 +101,28 @@ defmodule AeMdw.Dex do
     end
   end
 
+  @spec get_pair_pk(State.t(), Txs.txi(), Txs.txi(), Contracts.log_idx()) :: pubkey()
```

Comment:
Judging from the other similar function (`get_create_txi`) there should be a case, where the `ext_contract` is `nil`, that is not handled here. Also could we combine those two functions into one (or at least extract the specific internals into separate function), because it seems that changes into one of them should be reflected in the other one?

Reply:
The test case where `ext_contract` is `nil` should never happen because all swaps are going to be done via the proxy contract. Added the clause anyway for tests to pass. At some point I'd like to get rid of `get_create_txi` so that that is called separately (via `Origin.tx_index(pair_pk)`)